src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -4234,6 +4230,10 @@ namespace EnergyPlus::ZoneEquipmentManager {
                         returnNodeMassFlow = inletMassFlow;
                         thisZoneEquip.FixedReturnFlow(returnNum) = true;
                     }
+                    // if zone mass balance true, set to expected return flow
+                    if (DataHeatBalance::ZoneAirMassFlow.EnforceZoneMassBalance ) {
+                        returnNodeMassFlow = ExpTotalReturnMassFlow;
+                    }
```

Comment:
Resets the local variable to the expected zone total return air mass flow rate determined by the zone air mass flow balance calculation. When the zone mass conservation is active, this new code overrides the equation that sets the return mass flow rate to the supply inlet node flow rate.

Reply:
This may be fine when there's only one return node, but it won't work if there are multiple return nodes. `ExpTotalReturnMassFlow` is the sum of all return nodes serving the zone.

Reply:
This kind of adjustment needs to happen below in the `// Adjust return flows if greater than expected....` block starting at line 4280. Maybe it needs a different method to adjust the return flow when `EnforceZoneMassBalance` is true?

Reply:
I agree its the zone total return and may need a different method treatment. I tied a different approach but did not work and that is why I moved the code up the loop such the flow adjustment catches it if needed. This a work in progress.

Reply:
Another conversation that I will accept for the moment to keep things moving, but I want to make sure everyone is happy with this before 9.5 :smile: :smile: :smile: